doc/design/vic-machine/configure.md

```diff
@@ -0,0 +1,182 @@
+# VCH Configure (P0)
+This document defines the VCH configure command line, and the related changes in vic-machine, to make VCH configure process more reasonable.
+
+Here are the key points considerred in this Definition:
+- Batch Operation
+
+  We'd like to change VCH configurations more than one item at one time to make the command easy to use
+- Reuse all options used in vic-machine create
+
+  In vic-machine create, we already have 57 options to define VCH configurations, though not all of them can be updated, many of them should be able to be changed, and this number will grow over time, so we definitely do not want to introduce 3X that number of options to configure (configure means add/modify/remove)
+
+## Options
+First option is to configure all in one command
+
+### Implicit Add/Modify and Explicit Delete
+All existing vic-machine create options are supported in vic-machine configure command. The meaning of those options are as following:
+- anything not specified means no change
+- any option specified in configure command means replacement.
+  - if the old value is not specified during creation, this new value will be added
+  - if there is old value specified, no matter it's a list value or not, we'll replace the old value with the new one specified in configure command.
+
+    here are the possibilities to change list option ```--registry-ca```, if the create command option is ```--registry-ca ca1 --registry-ca ca2```
+	* ```--registry-ca ca2 --registry-ca ca3```, means new configuration will be [ca2, ca3], ca1 is deleted, and ca3 is added
+	* ```--registry-ca ca2```, means new configuration will be [ca2], ca1 is deleted
+	* ```--registry-ca ""```, means ca1 and ca2 are all deleted // Here the empty string value is required, and the representation in different os platform is different.
+	* nothing specified for registry-ca, then existing [ca1, ca2] is not changed
```

Comment:
I agree with this over my suggestion - it's completely consistent behaviour without trying to construct a special-case for addition. While this will make some common operations more awkward it's better than adding inconsistencies.